libs/gl-client/src/signer/mod.rs

```diff
@@ -754,37 +754,86 @@ impl Signer {
         VERSION
     }
 
-    /// Create a base64 encoded rune from the master rune of this signer. The
-    /// method enforces a restriction that contains an alternative with a
-    /// `pubkey` field and an equal condition set, e.g. `pubkey=02ea2d....`.
-    /// This is enforced to ensure that a developer does not hand out
-    /// unrestricted runes by accident.
+    /// Creates a base64 string called a rune which is used to authorize
+    /// commands on the node and to issue signatures from the signer. Each new
+    /// rune must contain a `pubkey` field that equals the public key that is
+    /// used to sign-off signature requests. Nobody can remove restrictions from
+    /// a rune.
+    ///
+    /// If a `rune` is supplied the restrictions are added to this rune. This
+    /// way one can invoke a rune that only allows for a subset of commands.
+    ///
+    /// `restrictions` is a vector of restrictions where each restriction itself
+    /// is a vector of one ore more alternatives.
+    ///
+    /// - =: passes if equal ie. identical. e.g. method=withdraw
+    /// - /: not equals, e.g. method/withdraw
+    /// - ^: starts with, e.g. id^024b9a1fa8e006f1e3937f
+    /// - $: ends with, e.g. id$381df1cc449605.
+    /// - ~: contains, e.g. id~006f1e3937f65f66c40.
+    /// - <: is a decimal integer, and is less than. e.g. time<1656759180
+    /// - \>: is a decimal integer, and is greater than. e.g. time>1656759180
+    /// - {: preceeds in alphabetical order (or matches but is shorter), e.g. id{02ff.
+    /// - }: follows in alphabetical order (or matches but is longer), e.g. id}02ff.
+    /// - #: a comment, ignored, e.g. dumb example#.
+    /// - !: only passes if the name does not exist. e.g. something!.  Every other operator except # fails if name does not exist!
+    ///
+    /// # Examples
+    /// This creates a fresh rune that is only restricted to a pubkey:
+    ///
+    /// `create_rune(None, vec![vec!["pubkey=000000"]])`
+    ///
+    /// "wjEjvKoFJToMLBv4QVbJpSbMoGFlnYVxs8yy40PIBgs9MC1nbDAmcHVia2V5PTAwMDAwMA"
+    ///
+    /// This adds a restriction to the rune, in this case a restriction that only
+    /// allows to call methods that start with "list" or "get", basically a
+    /// read-only rune:
+    ///
+    /// `create_rune("wjEjvKoFJToMLBv4QVbJpSbMoGFlnYVxs8yy40PIBgs9MC1nbDAmcHVia2V5PTAwMDAwMA", vec![vec!["method^list", "method^get"]])`
+    ///
     pub fn create_rune(
         &self,
-        unique_id: &str,
-        res: Vec<Restriction>,
+        rune: Option<&str>,
```

Comment:
Runes actually already have this feature `some_rune.add_restriction(futhark::Restriction)`. We could either change it upstream to accept a `&str` argument, this would then match your suggestion.